### PR TITLE
Search API should use Replica enums instead of strings

### PR DESCRIPTION
**Pull request by mweiden**
Fixes https://github.com/HumanCellAtlas/data-store/issues/772

### CODE REVIEW

File: dss/api/search.py [hunk 1]
```diff
@@ -31,24 +31,23 @@ def post(json_request_body: dict,
     es_query = json_request_body['es_query']
     per_page = PerPageBounds.check(per_page)
 
-    if replica is None:
-        replica = "aws"
+    replica_enum = Replica[replica] if replica else Replica.aws
```
Comment (ttung) [inline dss/api/search.py]: Strictly speaking, `if replica` is not the same as `if replica is None`

File: dss/config.py [hunk 2]
```diff
@@ -204,6 +204,9 @@ def get_es_index_name(index_type: ESIndexType,
         Returns the fully qualified name of an Elasticsearch index of documents for a given
         replica of a given type and shape.
         """
+        if not isinstance(replica, Replica):
```
Comment (ttung) [inline dss/config.py]: `assert isinstance(replica, Replica)`

File: dss/api/search.py (hunk 1 above)
Comment (mweiden) [reply dss/api/search.py]: ✅

File: dss/config.py (hunk 2 above)
Comment (mweiden) [reply dss/config.py]: ✅

**Pull request merged**